cmd/terramate/e2etests/run_test.go

```diff
@@ -441,11 +441,177 @@ stack-z
 			s.BuildTree(tc.layout)
 
 			cli := newCLI(t, s.RootDir())
-			args := []string{"plan", "run-order"}
-			if tc.changed {
-				args = append(args, "--changed")
-			}
-			assertRunResult(t, cli.run(args...), tc.want)
+			assertRunResult(t, cli.run("plan", "run-order"), tc.want)
+		})
+	}
+}
+
+func TestRunWants(t *testing.T) {
+	type testcase struct {
+		name   string
+		layout []string
+		wd     string
+		want   runExpected
+	}
+
+	for _, tc := range []testcase{
+		{
+			/* TODO(i4k): stack-a wants stack-a should error */
```

Comment:
should? @katcipis @mariux

Reply:
yeah I saw this and my first intuition  was that it should fail, it is kind odd for a stack to want itself. But on the other hand, if you want yourself the run set would be...yourself...and since you are not talking about order...seems OK, it is not like a cycle because since there is no ordering you are just adding things on a set...in this case adding yourself.

Reply:
could be a warning.. as we allow cyclic references this is not a hard fail?

Reply:
ok, added a warning.